Documentation/RepoCompose.md

```diff
@@ -0,0 +1,287 @@
+    }
+}
+```
+
+### File 
+
+Any file which is not a NuGet package should be listed as a file artifact.  These can be downloaded from the web or copied from local places on the hard drive.  Each type of file entry will have a name uniquely identifying the artifact and a kind property specifying the remainder of the properties:
+
+    - uri: a property named `"uri"` will contain an absolute Uri for the artifact.
+    - filesystem: a property named `"location"` will contain an OS specific file path for the artifact.
+
+Example: 
+
+``` json
+"file": {
+    "nuget.exe":  {
+        "kind": "uri",
+        "uri": "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe"
+    }, 
+    "run.exe": { 
+        "kind": "filesystem",
+        "location": "c:\\tools\\run.exe"
+    }
+}
+```
+
+## Samples
+
+### consumes
+
+``` json
+{
+    "os-all": {
+        "dependencies": {
+            "floating": { 
+                 "nuget": {
+                    "feeds": [
+                        { 
+                           "name": "core-clr",
+                           "value": "https://dotnet.myget.org/F/dotnet-coreclr/api/v3/index.json" 
+                        },
+                        {
+                            "name": "dotnet-core",
+                            "value": "https://dotnet.myget.org/F/dotnet-core/api/v3/index.json"
+                        }
+                    ],
+                    "packages" {
+                        "MicroBuild.Core": "0.2.0",
+                        "Microsoft.NETCore.Platforms": "1.0.1"
+                    }
+                },
+                "file": {
+                    "nuget.exe":  {
+                        "uri": "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe"
+                    }
+                }
+            },
+            "toolset": { 
+            },
+            "static": { 
+            }
+        },
+        "prereq": { 
+            "Microsoft Visual Studio": "2015",
+            "CMake" : "1.0"
+        } 
+    }
+}
+```
+
+### produces
+
+``` json
+{
+    "os-all": {
+        "nuget": {
+            "packages" {
+                "MicroBuild.Core": "0.2.0",
+                "Microsoft.NETCore.Platforms": "1.0.1"
+            }
+        },
+        "file": {
+            "nuget.exe": { } 
+        }
+    }
+}
+```
+
+## Implementation Stages
+
+Fully implementing the commands described in this document requires a decent amount of work.  The expectation is that it will be implemented in stages.  Each stage provides benefit to an existing tool or allows us to write a new tool for a major scenario:
+
+- Stage 1: NuGet build information.  NuGet packages are how the majority of repo artifacts are shared today.  Getting the NuGet section of the above commands implemented allows us to automate the majority of our composed builds.  This is really just a formalization of the commands repos already implement as a part of being inside the Maestro tool.  
```

Comment:
> This is really just a formalization of the commands repos already implement as a part of being inside the Maestro tool.  

This sentence is awkward, and I don't really know what it means.  Now that I read it a few times, I'd assume that it's supposed to be "commands **that** repos already implement", and not referring to some "commands repos" I am unaware of.